{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-10-13 12:49:50.026138+00:00\n"
     ]
    }
   ],
   "source": [
    "from datetime import datetime, timezone\n",
    "\n",
    "# Print the current UTC time\n",
    "print(datetime.now(timezone.utc))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Supabase client initialized: True\n",
      "https://mbkrvutzguuzqqnocgkp.supabase.co/storage/v1/object/public/selfies/filename.jpg?\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "import os\n",
    "import io\n",
    "import uuid  # For generating unique file names\n",
    "import traceback  # Importing traceback module\n",
    "from datetime import datetime, timezone\n",
    "from flask import Flask, request, jsonify\n",
    "from flask_cors import CORS\n",
    "from supabase import create_client, Client\n",
    "from face_recognition_face_recognition import FaceRecognitionFaceRecognition\n",
    "from face_recognition_interface import FaceRecognitionInterface\n",
    "from PIL import Image, ExifTags\n",
    "\n",
    "\n",
    "\n",
    "app = Flask(__name__)\n",
    "# Allow all origins for development purposes\n",
    "CORS(app, resources={r\"/*\": {\"origins\": \"*\"}})\n",
    "\n",
    "\n",
    "SUPABASE_URL = 'https://mbkrvutzguuzqqnocgkp.supabase.co'\n",
    "SUPABASE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im1ia3J2dXR6Z3V1enFxbm9jZ2twIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTcyODgyMDk4MiwiZXhwIjoyMDQ0Mzk2OTgyfQ.xbxaEgOkvH8PR0zi7E6o1xYrDlRT0dddr_849ATZESc'\n",
    "supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)\n",
    "print(f\"Supabase client initialized: {supabase is not None}\")  # Optional: For debugging\n",
    "\n",
    "selfie_url = supabase.storage.from_('selfies')\n",
    "print(supabase.storage.from_('selfies').get_public_url('filename.jpg'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (246647202.py, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn[31], line 1\u001b[0;36m\u001b[0m\n\u001b[0;31m    File \"/Users/coendewith/Documents/Projects/scanner/app.py\", line 267, in upload_image\u001b[0m\n\u001b[0m         ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "\n",
    "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "face_verification_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
